src/transaction/boot_sr.c

```diff
@@ -989,34 +995,49 @@ boot_make_temp_volume_fullname (char *temp_vol_fullname, VOLID temp_volid)
 }
 
 /*
- * boot_remove_unknown_temp_volumes () -
+ * boot_remove_unknown_temp_volumes () - to be called only upon server restart; unknown volumes means
+ *    volumes which are found existing (and should not be) vs the state that the server is at the
+ *    moment (ie: restart); performs the deletion in reverse order (aka LIFO) versus creation order
+ *    in order to keep the data consistent in case of abnormal server shutdown even when the boot
+ *    procedure is underway
  *
  * return: none
  */
 static void
 boot_remove_unknown_temp_volumes (THREAD_ENTRY * thread_p)
 {
-  VOLID temp_volid;
+  VOLID temp_volid = LOG_MAX_DBVOLID;
   char temp_vol_fullname[PATH_MAX];
 
-  if (boot_Db_parm->temp_last_volid == NULL_VOLID)
-    {
-      temp_volid = LOG_MAX_DBVOLID;
-    }
-  else
-    {
-      temp_volid = boot_Db_parm->temp_last_volid - 1;
-    }
+  assert (!BO_IS_SERVER_RESTARTED ());
+  /* logic below needs this invariant */
+  static_assert (LOG_MAX_DBVOLID < VOLID_MAX, "LOG_MAX_DBVOLID is expected to be less than VOLID_MAX");
 
-  for (; temp_volid > boot_Db_parm->last_volid; temp_volid--)
+  /* inspect volumes in descending order */
+  for (; temp_volid > LOG_DBFIRST_VOLID; temp_volid--)
     {
       boot_make_temp_volume_fullname (temp_vol_fullname, temp_volid);
```

Comment:
Does it make sense to memoize the results of `boot_make_temp_volume_fullname` in a stack-like struct and just pop them below? I noticed that this function might do some I/O when checking for `pwd`.

Reply:
@cristiarg I think it will have insignificant impact compared to the overall time to boot the database.